packages/itwinui-react/src/core/Dialog/DialogMain.tsx

```diff
@@ -227,8 +226,15 @@ export const DialogMain = React.forwardRef<HTMLDivElement, DialogMainProps>(
     return (
       <CSSTransition
         in={isOpen}
-        classNames='iui-dialog-animation'
+        classNames={{
+          enter: 'iui-dialog-animation-enter',
+          enterActive: 'iui-dialog-animation-enter-active',
+          enterDone: 'iui-dialog-visible',
+        }}
         timeout={{ exit: 600 }}
+        onEntered={() => {
+          setTimeout(() => dialogRef.current?.focus(), 300);
```

Comment:
@LostABike Are you sure this `setTimeout` is needed? In my testing it was working fine even without it, because `onEntered` is only called after the animation is done.

On a different note, you only want to call this code when the `setFocus` prop is set to `true`.

Also, previously this focus behavior was handled in useEffect, near line 87. I think you can remove much of that code now. The `previousFocusedElement` behavior could be handled through one of the `onExit` calbacks instead.

Reply:
Oh I see, so the focus works, but the animation is glitchy.

I believe this is because focused elements are automatically scrolled into view by the browser. You can override this behavior using [`preventScroll`](https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement/focus#parameters).

```js
dialogRef.current?.focus({ preventScroll: true })
```

Reply:
Oh man...today I learn 😅 Yea that'd do it

Reply:
Updated, removed `setTimeout` and the hook that was used to handle focusing, and added focus handling via callbacks.

Reply:
Okay I think this is what you mean. Updated.